Sources/VLCActionSheet.swift

```diff
@@ -0,0 +1,422 @@
+            collectionView.widthAnchor.constraint(equalTo: view.widthAnchor),
+            ])
+    }
+
+    fileprivate func setupBottomBackgroundView() {
+        NSLayoutConstraint.activate([
+            bottomBackgroundViewHeightConstraint,
+            bottomBackgroundView.widthAnchor.constraint(equalTo: view.widthAnchor)
+            ])
+    }
+
+    fileprivate func setupMainStackViewConstraints() {
+        NSLayoutConstraint.activate([
+            mainStackView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            mainStackView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            mainStackView.widthAnchor.constraint(equalTo: view.widthAnchor),
+            mainStackView.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+            ])
+    }
+
+    override func updateViewConstraints() {
+        super.updateViewConstraints()
+
+        if let presentingViewController = presentingViewController, let dataSource = dataSource {
+            collectionViewHeightConstraint.constant = CGFloat(dataSource.numberOfRows()) * cellHeight
+            maxCollectionViewHeightConstraint.constant = presentingViewController.view.frame.size.height / 2
+            collectionView.setNeedsLayout()
+            collectionView.layoutIfNeeded()
+        }
+    }
+
+    @available(iOS 11.0, *)
+    override func viewSafeAreaInsetsDidChange() {
+        bottomBackgroundViewHeightConstraint.constant = view.safeAreaInsets.bottom
+    }
+
+    // MARK: UIViewController
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.addSubview(backgroundView)
+        view.addSubview(mainStackView)
+
+        // wip: Currently adapting to new UI
+        //       option to hide cancelbutton
+        mainStackView.addArrangedSubview(headerView)
+        mainStackView.addArrangedSubview(collectionView)
+//        mainStackView.addArrangedSubview(cancelButton)
+        mainStackView.addArrangedSubview(bottomBackgroundView)
+
+        backgroundView.frame = UIScreen.main.bounds
+
+        setupMainStackViewConstraints()
+        setupCollectionViewConstraints()
+//        setupCancelButtonConstraints()
+        setuplHeaderViewConstraints()
+        setupBottomBackgroundView()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        mainStackView.isHidden = true
+        updateViewConstraints()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        // This is to avoid a horrible visual glitch!
+        mainStackView.isHidden = false
+
+        UIView.transition(with: backgroundView, duration: 0.2, options: .transitionCrossDissolve, animations: { [weak self] in
+            self?.backgroundView.isHidden = false
+            }, completion: nil)
+
+        let realMainStackView = mainStackView.frame
+
+        mainStackView.frame.origin.y += mainStackView.frame.origin.y
```

Comment:
why? what? please add an explaining comment.

Reply:
This was mainly for a small animation purposes, I will add some comments